python/pyspark/rdd.py

```diff
@@ -2509,7 +2509,7 @@ def pipeline_func(split, iterator):
         self._jrdd_deserializer = self.ctx.serializer
         self._bypass_serializer = False
         self.partitioner = prev.partitioner if self.preservesPartitioning else None
-        self.is_barrier = prev._is_barrier() or isFromBarrier
+        self.is_barrier = isFromBarrier or prev._is_barrier()
```

Comment:
I made this change into a separate PR - https://github.com/apache/spark/pull/23690